### ISSUE TITLE
Auto-unfocusing/blurring from the reference text input

### ISSUE THREAD

**Issue by nsunami** [libscie/ResearchEquals.com#326]
When I click the reference input box, the drop-down suggestion appears. But, after that, the drop-down closes, and I lose focus on the input box (then, I have to click the box again):

https://user-images.githubusercontent.com/17035406/153646521-464b64be-4f68-481e-8387-bd5054bf57df.mov

Edited: 
I was using Chrome @chartgerink

**Comment by chartgerink** [comment]
Thanks @nsunami for reporting this! 

Does this happen for other search boxes for you too? Also, for my information - what browser do you use?

**Comment by chartgerink** [comment]
Been investigating a bit and it seems like there's something up with the `ReactDOM.render` method in the `Autocomplete` component - it also shows this error everytime the dropdown closes.

![Screenshot 2022-02-14 at 15 04 45](https://user-images.githubusercontent.com/2946344/153879549-9d599606-5fb5-4414-ad47-f303704264ca.png)

This is the result of React@18, so will try to use [the documentation from React](https://github.com/reactwg/react-18/discussions/5) to see whether I can resolve the issue this way.

**Comment by nsunami** [comment]
This issue happens on the main search bar as well. 

https://user-images.githubusercontent.com/17035406/159473588-f1c981c4-f22c-4965-9b5e-9ee0c1888b6a.mov

(I'm curious to know how to solve this as I'm having the same issue with the PostReview app.)

**Comment by chartgerink** [comment]
@nsunami and I spent an hour trying to figure this out but didn't succeed in resolving this. We did succeed in figuring out:

* The warning I mentioned before is a separate issue most likely (#364 now)
* It is the hooks that are causing the component to be updated - which re-renders the autocomplete box, effectively closing it.

We did not have the issue on a page that had only the Autocomplete component but no hooks (see this [gist](https://gist.github.com/chartgerink/68c655114a75b5d2f1fbd403b05e9095)).

![](https://media.giphy.com/media/3owzVTAPpzlD3KV1Be/giphy.gif)
